src/decorator/ForeignKey.ts

```diff
@@ -0,0 +1,96 @@
+import { ObjectType } from "../common/ObjectType"
+import { getMetadataArgsStorage } from "../globals"
+import { ForeignKeyMetadataArgs } from "../metadata-args/ForeignKeyMetadataArgs"
+import { ForeignKeyOptions } from "./options/ForeignKeyOptions"
+import { ObjectUtils } from "../util/ObjectUtils"
+
+/**
+ * Creates a database foreign key. Can be used on entity property or on entity.
```

Comment:
```suggestion
 * Creates a database foreign key. Can be used on entity property or on entity.
 * Can create foreign key with composite columns when used on entity.
 * Warning! Don't use this with relations; relation decorators create foreign keys automatically.
```

Note I'm suggesting this change once, but you'll need to copy it to each overload signature below.